lib/hal/led.c

```diff
@@ -0,0 +1,27 @@
+#include <hal/led.h>
+#include <xboxkrnl/xboxkrnl.h>
+#include <assert.h>
+
+#define SMC_REG_LEDMODE 0x07
+#define SMC_REG_LEDSEQ  0x08
+
+void XSetAutoLED(void) {
+  HalWriteSMBusValue(0x20, SMC_REG_LEDMODE, FALSE, 0x00);
+}
+
+void XSetManualLED(unsigned int t1, unsigned int t2, unsigned int t3, unsigned int t4) {
+  HalWriteSMBusValue(0x20, SMC_REG_LEDMODE, FALSE, 0x01);
+
+  assert((t1 & ~0x11) == 0x00);
+  assert((t2 & ~0x11) == 0x00);
+  assert((t3 & ~0x11) == 0x00);
+  assert((t4 & ~0x11) == 0x00);
+
+  unsigned int colorScheme =
```

Comment:
`HalWriteSMBusValue` takes a `ULONG`; so it's probably best to use ULONG here, to avoid type issues